src/Function.cpp

```diff
@@ -299,11 +299,12 @@ class FreezeFunctions : public IRGraphVisitor {
     }
 };
 
-// A counter to use in tagging random variables
-std::atomic<int> rand_counter{0};
-
 }  // namespace
 
+// A counter to use in tagging random variables.
+// Note that this will be reset by Internal::reset_random_counters().
+std::atomic<int> random_variable_counter = 0;
```

Comment:
The only effect of this tag seems to be to give each compiled Pipeline a separate set of random numbers, in addition to each individual call to random_whatever being distinct. resetting the random counters in Module.cpp below seems like it would just totally undo this...

So if this counter is important, we'd better figure out why, because this PR breaks it.

Reply:
Disregard, this is a per-Func counter, not a per-Pipeline counter.

Reply:
Yes